FileOperationsAdvancedCases.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+using NUnit.Framework;
+using System.Diagnostics;
+
+namespace HelloWorld
+{
+    public class FileOperationsAdvancedCases
+    {
+
+        public void CreateFile(string fileName)
+        {
+            FileStream fs = File.Create(TestContext.CurrentContext.TestDirectory + fileName);
+            fs.Close();
+        }
+
+        public void CreateDirectory(string directoryName)
+        {
+            DirectoryInfo dir = new DirectoryInfo(TestContext.CurrentContext.TestDirectory);
+            DirectoryInfo subDir = dir.CreateSubdirectory(directoryName);
+        }
+
+        public void WriteToFile()
+        {
+            string path = TestContext.CurrentContext.TestDirectory + @"\FirstFile.txt";
+
+            if (!File.Exists(path))
+            {
+                File.WriteAllText(TestContext.CurrentContext.TestDirectory + @"\FirstFile.txt", "This is the first line!");
+            }
+            else if (File.Exists(path))
+            {
+                using (var tw = new StreamWriter(path, true))
```

Comment:
both WriteAllText and WriteLine are somegow insecure in terms of I/O operations. thus, I'd wrap the whole method body into the using(...) statement